packages/flutter/test/widgets/form_test.dart

```diff
@@ -940,4 +940,61 @@ void main() {
     fieldKey.currentState!.reset();
     expect(fieldKey.currentState!.hasInteractedByUser, isFalse);
   });
+
+testWidgets('Validator is nullified and error text behaves accordingly',
+    (WidgetTester tester) async {
+  final GlobalKey<FormState> formKey = GlobalKey<FormState>();
+  bool useValidator = false;
+
+  String? errorText(String? value) {
+    if (value == null || value.isEmpty) {
+      return 'test_error';
+    }
+    return null;
+  }
+
+  Widget buildTestApp() {
+    return MaterialApp(
+      home: MediaQuery(
+        data: const MediaQueryData(),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Center(
+            child: Material(
+              child: Form(
+                key: formKey,
+                child: TextFormField(
+                  validator: useValidator ? errorText : null,
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  // Start with no validator
+  await tester.pumpWidget(buildTestApp());
+
+  // Enter empty text, expect no error since validator is null
+  await tester.enterText(find.byType(TextFormField), '');
+  formKey.currentState!.validate();
+  await tester.pump();
+  expect(find.text('test_error'), findsNothing);
+
+  // Now use the validator
+  useValidator = true;
+  await tester.pumpWidget(buildTestApp());
```

Comment:
Nit: Would it be any better to do this test with setState instead of re-pumping an entire widget tree?  I'm not sure if it would really matter here or not, but it might be more reassuring.

You could use StatefulBuilder [like this](https://github.com/flutter/flutter/blob/00a99d3bc09bed0433319f05f5161557d11039f8/packages/flutter/test/widgets/form_test.dart#L471).